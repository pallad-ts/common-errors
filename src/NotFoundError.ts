export class NotFoundError<T = { [key: string]: any }> extends Error {
    constructor(message: string, readonly references?: T) {
        super(message);
        this.message = message;
        this.name = 'NotFoundError';
        Object.freeze(this);
    }

    static entity(name: string, reference: { [key: string]: any }) {
        const referenceString = Object.keys(reference)
            .reduce((result, key) => {
                return result.concat([`${key}: ${reference[key]}`]);
            }, [] as string[])
            .join(', ');
        new NotFoundError(`${name} not found${referenceString ? ` ${referenceString}` : ''}`)
    }
}