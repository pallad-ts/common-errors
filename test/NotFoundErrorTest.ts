import {NotFoundError} from "@src/NotFoundError";

describe('NotFoundError', () => {
    describe('creating for entity', () => {
        it('without references', () => {
            const error = NotFoundError.entity('User');

            expect(error.message)
                .toMatchSnapshot();
        });

        it('with references', () => {
            const error = NotFoundError.entity('User', {id: 10});

            expect(error.message)
                .toMatchSnapshot();
        });
    });
})