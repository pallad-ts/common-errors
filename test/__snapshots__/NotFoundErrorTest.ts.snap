// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundError creating for entity with references 1`] = `"User not found - id: 10"`;

exports[`NotFoundError creating for entity without references 1`] = `"User not found"`;
